From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct mlx5_modify_qp_mbox_in
Patch-mainline: never, kabi

In 3.12.63, commit c70a6ed3121a631ed40cdce043551640b013690a (IB/mlx5:
Fix MODIFY_QP command input structure), upstream commit
e3353c268b06236d6c40fa1714c114f21f44451c renamed two reserved fields
in struct mlx5_modify_qp_mbox_in. It made the kABI checker to report
the structure has changed.

The structure actually did not change, so pretend that the usual way,
by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/mlx5/qp.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/mlx5/qp.h
+++ b/include/linux/mlx5/qp.h
@@ -485,9 +485,15 @@ struct mlx5_destroy_qp_mbox_out {
 struct mlx5_modify_qp_mbox_in {
 	struct mlx5_inbox_hdr	hdr;
 	__be32			qpn;
+#ifdef __GENKSYMS__
+	u8			rsvd1[4];
+	__be32			optparam;
+	u8			rsvd0[4];
+#else
 	u8			rsvd0[4];
 	__be32			optparam;
 	u8			rsvd1[4];
+#endif
 	struct mlx5_qp_context	ctx;
 };
 
